### ISSUE TITLE
🦾 Organization issue project workflows

### ISSUE THREAD

**Issue by JFWooten4** [WhyDRS/.github#8]
It looks like the projects tab only allows for the basic template actions. Unfortunately, I cannot duplicate the DB add or include more repositories therein by my understanding. It [seems like](https://chatgpt.com/share/671d7a8f-7e30-8010-8f13-fc7fd30255d6) migrating into a [new workflow](https://github.com/WhyDRS/.github/blob/main/workflows/Project_Issue_Board.yml) ought best help bring everything into one place (with categories?).

cc requester @JamesAlfonse

**Comment by JamesAlfonse** [comment]
Setting up a workflow using GraphQL, Personal Access Tokens and Workflows would probably be the easiest way to do it. A Github app could also be made, but that would require setting up a server. I think this may take a good amount of time, and it may just be easier to add issues manually to the project for now. I continued the conversation you started below, let me know what your thoughts are.

https://chatgpt.com/share/671dc643-e8fc-800c-9786-d3a9585ea26a

**Comment by JFWooten4** [comment]
@JamesAlfonse I love the idea of using Personal Access Tokens for accountability (GPT gave me a similar response, and your chat flow looks quite logical). It's certainly a new thing for me, and perhaps something to take a step back from to ask how we are using the projects board. I've been setting them up for much of [my own](https://github.com/users/JFWooten4/projects/3) work recently.

The more I think about it, the more it sounds like we're (possibly) giving people a list of things and being like "this is what needs to happen, go do it." Maybe you see it as something more organizational, and by all means let me know your thoughts on the potential here. But what I'm trying to say is that I think it's best when people like you and I decide independently what to work on, where to assign things in our own versions of projects, and so forth.

That train of logic is why many of my own projects are specifically written for my own understanding alone, using convenient shorthand that still keeps everything organized around outstanding GitHub Issues/PRs (generally that I've created or am collaborating on with someone else). But when we [delegate those choices](https://assets.sbnation.com/assets/1074301/Valve_Handbook_LowRes.pdf#page=27) to a "robot overlord" (as dramatic as I'm making the automation bot sound lol), I just think it removes a lot of the fun that comes with discovering a unique challenge I'd personally like to solve&mdash;and then _my own_ journey to go and fix it.

**Comment by JamesAlfonse** [comment]
> @JamesAlfonse I love the idea of using Personal Access Tokens for accountability (GPT gave me a similar response, and your chat flow looks quite logical). It's certainly a new thing for me, and perhaps something to take a step back from to ask how we are using the projects board. I've been setting them up for much of [my own](https://github.com/users/JFWooten4/projects/3) work recently.
> 
> The more I think about it, the more it sounds like we're (possibly) giving people a list of things and being like "this is what needs to happen, go do it." Maybe you see it as something more organizational, and by all means let me know your thoughts on the potential here. But what I'm trying to say is that I think it's best when people like you and I decide independently what to work on, where to assign things in our own versions of projects, and so forth.
> 
> That train of logic is why many of my own projects are specifically written for my own understanding alone, using convenient shorthand that still keeps everything organized around outstanding GitHub Issues/PRs (generally that I've created or am collaborating on with someone else). But when we [delegate those choices](https://assets.sbnation.com/assets/1074301/Valve_Handbook_LowRes.pdf#page=27) to a "robot overlord" (as dramatic as I'm making the automation bot sound lol), I just think it removes a lot of the fun that comes with discovering a unique challenge I'd personally like to solve—and then _my own_ journey to go and fix it.

I totally get where you're coming from. It's challenging to motivate people to take on tasks that best match their skills, especially when automation is involved with those tasks. My vision for the projects board, like you mentioned, is more for organizational purposes. I want to make it easier for new participants to have a single page where they can see where they can contribute.

I still believe in the potential of this approach, but maybe we can tweak the system after we centralize the issues. Instead of automatically assigning categories, how about we structure the columns based on each repository? Since repositories are inherently organized, the automation could assign issues accordingly. Plus, we could organize these columns in a way that minimizes bias, such as sorting them alphabetically or by repository creation date, rather than manually assigning the positions, which could unintentionally highlight certain issues.

I really appreciate your insights and your willingness to think outside the box about potential challenges. I'm also open to other ideas for organizing issues or even shifting our focus elsewhere for now and revisiting this strategy later.

**Comment by JFWooten4** [comment]
> I still believe in the potential of this approach, but maybe we can tweak the system after we centralize the issues. Instead of automatically assigning categories, how about we structure the columns based on each repository? Since repositories are inherently organized, the automation could assign issues accordingly.

Sounds like a smart plan to me! 🧠 I'm happy to configure you as an org Admin to handle these configuration logistics.[^tm] That sound acceptable to other admin @tehchives or at all helpful/desired, James?

[^tm]: Temporarily, at least. I know we're in the process of detailing the roles and [custody](https://github.com/Taking-Stock/RSSfeed/issues/4) things in the new DAO docs.

**Comment by tehchives** [comment]
Absolutely, it'd be very welcome. If James is agreeable. 

I like the vision to have the tasks be organized by something empirical and separate from individual priority or bias.

It really is tough to get people to engage, and I think part of that is phrasing. When operating as a volunteer collective and developing this open source project from the ground up we've needed to wait for people to have the time and interest to contribute on their own, in the way they choose, aligned with their values and desires for the future.

Maybe as we grow and once we have operating funds we can organize some tasks (or nominate them) for users whose efforts are more specifically sanctioned and supported by those funds, but I think that's something for much later.

Another thought, perhaps we could set up a viewing area for tasks in the vein of John's dao.whydrs.org and call it something like tasks.whydrs or contribute.whydrs or volunteer.whydrs, etc etc, and maintain representation of issues in progress there. This might break down the barriers for contribution even further. I am not sure if that conflicts with the per-repo issue organization though.

**Comment by JFWooten4** [comment]
Great idea at the end. Off the cuff, I think it compliments the per-repo organization because now it's another parameter you can filter for if you have an idea on the topic you want to work on generally. E.g., a writer (traditionally) could stick with Comments, Petitions, or Docs; whereas a seasoned dev might immediately hover towards the Database, Historic LLM, or a flashy homepage.

I think it's about making things easier to understand. Accessibility drastically lowers how long it takes someone to understand what an impact they might create. E.g. there were a ton of local things on the ballot this election that I had no clue about, and even just a simple UI `info` 2-sentence explainer in plain English would've materially helped my decision-making.

**Comment by JamesAlfonse** [comment]
@JFWooten4 @tehchives Yes, that works for me. Also loving the idea of having a subdomain(s) set up for this! Would love to see how that could be fleshed out. Are you envisioning something more than just a redirect to the projects page on Github?

**Comment by tehchives** [comment]
Yes, at least in UI. I think (and forgive me for this) but the github UI is still limiting, despite how intuitive it can seem. A frontend page with a simple domain describing what we're looking for might incite more contribution! I am hopeful and optimistic this is the case. 

This probably medium term wraps into the 'apes guide to github' idea.

**Comment by JFWooten4** [comment]
I second Chives' semantics and added James as discussed.[^2fa]

[^2fa]: Due also to establishment of 2FA.

**Comment by JamesAlfonse** [comment]
The GH app adds new issues daily, but I would like to circle back to potentially using a webhook with the GH app to instantaneously add issues. Also, I adjusted the default view to include groupings and splicings by repository. Since it's using some of the built-in functionality instead of a script, there's less opportunity for something to break. Looking for some feedback on that. Ihttps://github.com/orgs/WhyDRS/projects/3

**Comment by JFWooten4** [comment]
Thanks for all the hard work here and in whydrs/database#30 addressing some of these crucial items @JamesAlfonse! Love the inspirational progress and was pleasantly surprised by the organized view of Projects/3. It's stellar to have a single place users can visit and search by keyword for something that aligns with their skills 🧠

**Comment by JamesAlfonse** [comment]
Adding this screenshot for potential statuses to include in the project.

![Image](https://github.com/user-attachments/assets/481374af-83bb-44c5-a203-e127bad20be5)

**Comment by JamesAlfonse** [comment]
@tehchives @JFWooten4 How do you guys feel about having the following Issue statuses: Backlog, To do, In progress, Review, Testing, and Blocked? The Done column would be removed, any closed issues would fall off the Project and a new Project could be made for all closed issues. The default category would be "Backlog" which would be assigned to all new issues. I believe this issue can then be closed, and a new one created for the other Project if that works for everyone.

**Comment by tehchives** [comment]
I think it sounds excellent! Definitions you laid out above keep it pretty clear as well.

**Comment by JFWooten4** [comment]
@JamesAlfonse What would fall under blocked?

**Comment by JamesAlfonse** [comment]
@JFWooten4 That column would be for issues that aren't necessarily closed, but rather issues that either don't fall into the scope of whyDRS or issues that the DAO chooses not to currently work on. The word "Blocked" may be too strong of a term to use here, perhaps there's a better header that works? Or would it make more sense leaving out that column entirely?
Or instead, it can be listed in the future project that includes closed issues.

**Comment by JFWooten4** [comment]
@JamesAlfonse One thing I've seen previously is categorizing these as `ideas` which could be worked on in the future, but they are explicitly not in the backlog. This aspect just stood out to me because I've had too many PRs shut down directorily by Microsoft staff because it isn't "in their agenda." Sort of reminds me of the Wright ultimatum to take down "his whitepaper" because you need to do things according to centralized interests.

One of the interesting things about the DAO structure imo: anybody can propose something and then build it themselves if they truly believe in the idea's potential.[^ot]

[^ot]: Or hire someone to build it, if the topic is outside their skillset. And ofc for those without the current resources, there is at least somewhere to reference to others when disucssing and trying to gain support. For example, I've been campaigning for months both on GitHub, privately, and Discord trying to get stellar/stellar-docs#797.

**Comment by JamesAlfonse** [comment]
@JFWooten4 I see where you're coming from. Thanks for the added context. I like "Ideas", but I also think it can be improved upon with something more specific and actionable. How about "Ideas to Explore"?

**Comment by JamesAlfonse** [comment]
Thanks for the input everyone. I'm going to close this issue for now. Columns are set up [here](https://github.com/orgs/WhyDRS/projects/3/views/1), and as discussed in the biweekly chat the Done column will remain for now until it gets unmanageable. New issues are added daily and assigned a Backlog status. Webhooks pointing from Github back to github require setting up a third-party service, which could be explored in the future if needed.

**Comment by JamesAlfonse** [comment]
I renamed the project to Issues Tracker, which I believe sounds better/more accurate.
